From: Boyuan Yang <byang@debian.org>
Date: Fri, 1 Nov 2019 14:49:40 -0400
Subject: 64-arphic-uming.conf: Weaken priority and remove modification to
 DejaVu Serif

After Ubuntu 19.04 and Debian 10, both Ubuntu and Debian have decided to
prefer Noto CJK fonts. This adjustment to 64-arphic-uming.conf changes
the strong "prepend" relationship of fonts to "append" so that to make
sure that arphic fonts do not take priority.

Besides, the modification to the priority of DejaVu Serif are removed
as well to solve the issue as originally raised in
https://bugs.launchpad.net/ubuntu/+source/fonts-arphic-uming/+bug/1834561 .

Part of downstream Ubuntu's 65-arphic-uming.conf is merged inside to
make uming fonts available for serif and monospace (sans-serif
excluded).

Signed-off-by: Boyuan Yang <byang@debian.org>
---
 64-arphic-uming.conf | 90 +++++++++++++++-------------------------------------
 1 file changed, 25 insertions(+), 65 deletions(-)

diff --git a/64-arphic-uming.conf b/64-arphic-uming.conf
index b20b2ad..ed93b0a 100644
--- a/64-arphic-uming.conf
+++ b/64-arphic-uming.conf
@@ -1,6 +1,26 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
+	<!-- Generic name aliasing -->
+	<alias>
+		<family>serif</family>
+		<accept>
+			<family>AR PL UMing CN</family>
+			<family>AR PL UMing HK</family>
+			<family>AR PL UMing TW</family>
+		</accept>
+	</alias>
+
+	<alias>
+		<family>monospace</family>
+		<accept>
+			<family>AR PL UMing CN</family>
+			<family>AR PL UMing HK</family>
+			<family>AR PL UMing TW</family>
+		</accept>
+	</alias>
+
+	<!-- language specific settings -->
 	<match target="pattern">
 		<test qual="any" name="lang" compare="contains">
 			<string>zh-cn</string>
@@ -8,7 +28,7 @@
 		<test qual="any" name="family">
 			<string>serif</string>
 		</test>
-		<edit name="family" mode="prepend" binding="strong">
+		<edit name="family" mode="append">
 			<string>AR PL UMing CN</string>
 		</edit>
 	</match>
@@ -20,32 +40,8 @@
 		<test qual="any" name="family">
 			<string>serif</string>
 		</test>
-		<edit name="family" mode="prepend" binding="strong">
-			<string>AR PL UMing CN</string>
-		</edit>
-	</match>
-
-	<match target="pattern">
-		<test qual="any" name="lang" compare="contains">
-			<string>zh-cn</string>
-		</test>
-		<test name="family">
-			<string>AR PL UMing CN</string>
-		</test>
-		<edit name="family" mode="append" binding="strong">
-			<string>DejaVu Serif</string>
-		</edit>
-	</match>
-
-	<match target="pattern">
-		<test qual="any" name="lang" compare="contains">
-			<string>zh-sg</string>
-		</test>
-		<test name="family">
+		<edit name="family" mode="append">
 			<string>AR PL UMing CN</string>
-		</test>
-		<edit name="family" mode="append" binding="strong">
-			<string>DejaVu Serif</string>
 		</edit>
 	</match>
 
@@ -56,7 +52,7 @@
 		<test qual="any" name="family">
 			<string>serif</string>
 		</test>
-		<edit name="family" mode="prepend" binding="strong">
+		<edit name="family" mode="append">
 			<string>AR PL UMing HK</string>
 		</edit>
 	</match>
@@ -68,35 +64,11 @@
 		<test qual="any" name="family">
 			<string>serif</string>
 		</test>
-		<edit name="family" mode="prepend" binding="strong">
+		<edit name="family" mode="append">
 			<string>AR PL UMing HK</string>
 		</edit>
 	</match>
 
-	<match target="pattern">
-		<test qual="any" name="lang" compare="contains">
-			<string>zh-hk</string>
-		</test>
-		<test name="family">
-			<string>AR PL UMing HK</string>
-		</test>
-		<edit name="family" mode="append" binding="strong">
-			<string>DejaVu Serif</string>
-		</edit>
-	</match>
-
-	<match target="pattern">
-		<test qual="any" name="lang" compare="contains">
-			<string>zh-mo</string>
-		</test>
-		<test name="family">
-			<string>AR PL UMing HK</string>
-		</test>
-		<edit name="family" mode="append" binding="strong">
-			<string>DejaVu Serif</string>
-		</edit>
-	</match>
-
 	<match target="pattern">
 		<test name="lang" compare="contains">
 			<string>zh-tw</string>
@@ -104,21 +76,9 @@
 		<test qual="any" name="family">
 			<string>serif</string>
 		</test>
-		<edit name="family" mode="prepend" binding="strong">
+		<edit name="family" mode="append">
 			<string>AR PL UMing TW</string>
 		</edit>
 	</match>
 
-	<match target="pattern">
-		<test name="lang" compare="contains">
-			<string>zh-tw</string>
-		</test>
-		<test name="family">
-			<string>AR PL UMing TW</string>
-		</test>
-		<edit name="family" mode="append" binding="strong">
-			<string>DejaVu Serif</string>
-		</edit>
-	</match>
-
 </fontconfig>
